src/state.rs

```diff
@@ -1,52 +1,132 @@
-use super::gui::GUI;
+use super::gui::Gui;
+// Move these to preludes later
+use super::memory::MemoryManagers;
+use memory::game_engine::il2cpp::{Image, Module};
 use memory::process::Process;
 use memory::process_list::ProcessList;
+use std::time::Instant;
+use crate::gui::helpers::GuiHelper;
+use egui_dock::DockState;
+
+pub struct StateDebug {
+    pub last_update: Instant,
+    pub last_memory_update: Instant,
+}
+
+pub struct StateGui {
+    pub helpers: Vec<Box<dyn GuiHelper>>,
+    pub dock_state: DockState<String>
+}
 
 pub struct State {
-    pub process: Option<Process>,
-    pub module: u64,
+    pub context: StateContext,
     pub process_list: ProcessList,
+    pub memory_managers: MemoryManagers,
+    pub debug: StateDebug,
+    pub gui: StateGui
+}
+
+pub struct StateContext {
+    pub process: Option<Process>,
+    pub module: Option<Module>,
+    pub image: Option<Image>,
 }
 
 impl State {
-    pub fn new(cc: &eframe::CreationContext<'_>) -> Self {
```

Comment:
Note, the comment is removed here, so it's not clear what the `_cc` parameter is for. It doesn't implement a trait either, so it seems to be not needed here.
You mentioned persistence in the chat before and it's commented out in `Cargo.toml`, but I guess you didn't get it working.